exercises/collatz-conjecture/test/collatz_conjecture_test.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:test/test.dart';
+import 'package:collatz_conjecture/collatz_conjecture.dart';
+
+void main() {
+  final collatzConjecture = new CollatzConjecture();
+
+  group('CollatzConjecture', () {
+    test("zero steps for one", () {
+      final int result = collatzConjecture.steps(1);
+      expect(result, equals(0));
+    }, skip: false);
+
+    test("divide if even", () {
+      final int result = collatzConjecture.steps(16);
+      expect(result, equals(4));
+    }, skip: true);
+
+    test("even and odd steps", () {
+      final int result = collatzConjecture.steps(12);
+      expect(result, equals(9));
+    }, skip: true);
+
+    test("large number of even and odd steps", () {
+      final int result = collatzConjecture.steps(1000000);
+      expect(result, equals(152));
+    }, skip: true);
+
+    test("zero is an error", () {
+      expect(() => collatzConjecture.steps(0), throwsArgumentError);
```

Comment:
Can you have the expects check that the error message that's part of the Error says "Only positive numbers are allowed". That's in the canonical-data.json.

Reply:
Will do, I'll make the update tonight!

Reply:
@Stargator Updated, let me know if this is the right way of doing this. Couldn't find a better way.

Reply:
@jvarness and @amscotti I'm struggling with this.

Yes, in this case the test suite has defined a predicate and the student would not need to understand it, just accept that it works.

Since collatz-conjecture is otherwise so simple and would then become the first exercise to introduce exception handling, I would advocate for a brief description of what a predicate is and why it is there as well as a link to the Dart docs on predicates.

Just so student who are curious can learn more about it.

Reply:
+1, adding that as a comment in the tests would be helpful I think.

Reply:
Okay, good. Sorry for the delay @amscotti

Reply:
@jvarness @Stargator I have added a comment into the tests file but please feel free to let me know if something else would be better to say. More than happy to change it to make it clearer.